baybe/targets/enum.py

```diff
@@ -0,0 +1,29 @@
+"""Target-related enumerations."""
+
+from enum import Enum
+
+
+class TargetMode(Enum):
+    """Available modes for targets."""
+
+    MIN = "MIN"
+    """The target is to be minimized."""
+
+    MAX = "MAX"
+    """The target is to be maximized."""
+
+    MATCH = "MATCH"
+    """The target should be close to a given value."""
+
+
+class TargetTransformMode(Enum):
```

Comment:
For me, having both an enum and the corresponding functions feels like boilerplate. Shouldn't there be a possibility for not requiring us to have both an enum here and the transform written in `targets.transforms.py`? This solution requires to manually keep these two different files in sync which I feel unhappy about.

Reply:
Hm, can you propose an alternative 🤔 Not sure if there is much room for reduction, but I'd be glad if you can come up with a more condensed approach. 

The issues that I see (but perhaps there is a workaround):
- Somewhere in the code, we must define (in one way or another) what is the mapping between modes and allowed transformations. This information simply does not come from the enum itself and requires additional specifications.
- Instead of specifying the mapping from modes to transforms, I could specify it directly from modes to transform callables, getting rid of the addtional function that implements the selection logic. However, then I lose the high-level representation of these functions and run into difficulties at other places, e.g. I can no longer do `value not in _VALID_TRANSFORM_MODES[self.mode]` ...

Let me know if you have a concept in mind that works and gets rid of one of the components 👍🏼

Reply:
Wouldn't it be possible to have an abstract base class for `TRANSFORMS_FUNCTIONS` and then have classes for the individual functions?

Reply:
I think this idea is valid, but not for this PR. The same holds for the different Objective modes, once we have more objective modes they might better be turned into classes. The target transformation classes idea is better placed in the same basket rather than here

Reply:
Ok, so I propose the following: We move this discussion here to the discussion part (in the way of "Let's discuss this at a later point in time"), copying @AdrianSosic arguments/mentioned issues. Once this is done I am happy to accept this PR :)